### PR TITLE
#13 Fixes only keep one less backup than releases kept

### PR DESCRIPTION
**Pull request by daften**
Also checks to make sure at least one release dir is kept

### CODE REVIEW

File: src/AbstractRoboFile.php [hunk 1]
```diff
@@ -1033,11 +1033,12 @@ protected function clearOpCacheTask($worker, AbstractAuth $auth, $remote)
     protected function cleanDirsTask($worker, AbstractAuth $auth, $remote)
     {
         $currentProjectRoot = $remote['currentdir'] . '/..';
+        $cleandirLimit = isset($remote['cleandir_limit']) ? max(1, $remote['cleandir_limit']) : '';
         return $this->taskSsh($worker, $auth)
                 ->remoteDirectory($currentProjectRoot, true)
                 ->timeout(30)
-                ->exec('vendor/bin/robo digipolis:clean-dir ' . $remote['releasesdir'] . (isset($remote['cleandir_limit']) ? ':' . $remote['cleandir_limit'] : ''))
-                ->exec('vendor/bin/robo digipolis:clean-dir ' . $remote['backupsdir'] . (isset($remote['cleandir_limit']) ? ':' . $remote['cleandir_limit'] : ''));
+                ->exec('vendor/bin/robo digipolis:clean-dir ' . $remote['releasesdir'] . (isset($remote['cleandir_limit']) ? ':' . $cleandirLimit : ''))
```
Comment (matthijs-va) [inline src/AbstractRoboFile.php]: I suggest you use `!empt($cleandirLimit)` instead of `isset(...)`

**Review by matthijs-va — CHANGES_REQUESTED** [review]

File: src/AbstractRoboFile.php (hunk 1 above)
Comment (zero2one) [reply src/AbstractRoboFile.php]: `!empty()`

Comment (dubgeiser) [reply src/AbstractRoboFile.php]: Why have the check _again_ in the exec() call, since it was already done above when initializing $cleandirLimit?

Comment (daften) [reply src/AbstractRoboFile.php]: 0 is also a valid value, with this we would exclude it

Comment (daften) [reply src/AbstractRoboFile.php]: @dubgeiser because if it isn't set, we use the defaults set higher up the chain.

**Review by matthijs-va — APPROVED** [review]

**Review by zero2one — APPROVED** [review]

File: src/AbstractRoboFile.php [hunk 2]
```diff
@@ -1033,11 +1033,12 @@ protected function clearOpCacheTask($worker, AbstractAuth $auth, $remote)
     protected function cleanDirsTask($worker, AbstractAuth $auth, $remote)
     {
         $currentProjectRoot = $remote['currentdir'] . '/..';
+        $cleandirLimit = isset($remote['cleandir_limit']) ? max(1, $remote['cleandir_limit']) : '';
         return $this->taskSsh($worker, $auth)
                 ->remoteDirectory($currentProjectRoot, true)
                 ->timeout(30)
-                ->exec('vendor/bin/robo digipolis:clean-dir ' . $remote['releasesdir'] . (isset($remote['cleandir_limit']) ? ':' . $remote['cleandir_limit'] : ''))
-                ->exec('vendor/bin/robo digipolis:clean-dir ' . $remote['backupsdir'] . (isset($remote['cleandir_limit']) ? ':' . $remote['cleandir_limit'] : ''));
+                ->exec('vendor/bin/robo digipolis:clean-dir ' . $remote['releasesdir'] . (!empty(cleandirLimit) ? ':' . $cleandirLimit : ''))
```
Comment (Jelle-S) [inline src/AbstractRoboFile.php]: s/cleandirLimit/$cleandirLimit/

File: src/AbstractRoboFile.php [hunk 3]
```diff
@@ -1033,11 +1033,12 @@ protected function clearOpCacheTask($worker, AbstractAuth $auth, $remote)
...
+                ->exec('vendor/bin/robo digipolis:clean-dir ' . $remote['backupsdir'] . (!empty(cleandirLimit) ? ':' . $cleandirLimit-1 : ''));
```
Comment (Jelle-S) [inline src/AbstractRoboFile.php]: s/cleandirLimit/$cleandirLimit/

**Review by Jelle-S — CHANGES_REQUESTED** [review]

**Review by Jelle-S — APPROVED** [review]

**Pull request merged**